disperser/dataapi/metrics.go

```diff
@@ -206,6 +206,11 @@ func (g *Metrics) UpdateSemverCounts(semverData map[string]*semver.SemverMetrics
 	}
 }
 
+// RegisterCollector registers a prometheus collector with the metrics registry
+func (g *Metrics) RegisterCollector(collector prometheus.Collector) {
+	g.registry.MustRegister(collector)
+}
```

Comment:
this is just a nit (no need to change) as I haven't looked at the whole metrics code and am not 100% sure, but typically I feel like a Metrics struct should only deal with metrics, not with a registry. Aka it should get the registry injected at construction time, and not expose a method like this.

Reply:
6aae9802fb8446907ac3b7d5cedc26e6e08e8842